sdks/java/core/src/main/java/org/apache/beam/sdk/util/WindowingStrategy.java

```diff
@@ -265,4 +271,37 @@ public int hashCode() {
     return Objects.hash(triggerSpecified, allowedLatenessSpecified, modeSpecified,
         windowFn, trigger, mode, allowedLateness, closingBehavior);
   }
+
+  /**
+   * An {@link OutputTimeFn} that uses {@link WindowFn#getOutputTime} to assign initial timestamps
+   * but then combines and merges according to a given {@link OutputTimeFn}.
```

Comment:
I think it's worth expanding this comment a bit:
 - we give the windowfn the chance to move hold forward to avoid later sliding windows holding up progress for earlier.
 - we the given the outputtimefn the chance to move forward again.
 - the outputtimefn retains responsibility for merging hold timestamps